src/kdbindings/node.h

```diff
@@ -223,6 +223,64 @@ class PropertyNode : public NodeInterface<PropertyType>
     mutable bool m_dirty;
 };
 
+template<typename PropertyType>
+class ConstPropertyNode : public NodeInterface<PropertyType>
```

Comment:
Does this really have to be its own type? Can you try to just make the `PropertyNode` take a `const Property<T>&` instead?

Reply:
Alright that would be interesting, we would need to ensure that signal connections do not inadvertently allow modifications to properties that should be immutable. This might require conditional logic based on the constness of the property, we can try to have an another template argument something like `IsConst` to the `PropertyNode`  and handle the members internally based on constness and we can then use it in this way

```
// PropertyNode<MyType, true> for const properties
// PropertyNode<MyType, false> for non-const properties
```

what do you think?